crates/ruff_db/src/vendored.rs

```diff
@@ -273,12 +299,18 @@ fn normalize_vendored_path(path: &VendoredPath) -> NormalizedVendoredPath {
             camino::Utf8Component::Normal(part) => normalized_parts.push(part),
             camino::Utf8Component::CurDir => continue,
             camino::Utf8Component::ParentDir => {
-                normalized_parts.pop();
+                if normalized_parts.pop().is_none() {
+                    return Err(InvalidVendoredPathError::EscapeFromZipFile);
+                }
+            }
+            unsupported => {
+                return Err(InvalidVendoredPathError::UnsupportedComponent(
+                    unsupported.to_string(),
+                ))
```

Comment:
Am I correct that this is only to handle prefixes? I guess the problem we run into here is that `camino::Utf8Path` path parsing is platform-dependent. I'm not sure how to solve this best but I'm also not sure if this special case is worth returning an `Error` (we control the path creation, it's not that they're provided from externally)

Reply:
Or a `camino::Utf8Component::RootDir`... which I don't think is possible in the middle of a file, but that can't be validated using the type system